FrontEnd/src/main/java/resourceDisplay/AssetsRepository.java

```diff
@@ -9,93 +10,84 @@
  */
 public class AssetsRepository {
 
-    public List<String> getFilterList(DatabaseClient dbClient, String workspaceId, String filterType) {
+    public List<String> getFilterList(DatabaseClient dbClient, String workspaceId,
+                                      String filterType) {
         List<String> filterList = new ArrayList<>();
-        String statementString = "SELECT DISTINCT ";
-        statementString += filterType;
-        statementString += " FROM Main_Assets WHERE workspaceId = @workspaceId";
+        String statementString = String.format("SELECT DISTINCT %s FROM Main_Assets WHERE " +
+                "workspaceId = @workspaceId ORDER BY %s", filterType, filterType);
         Statement statement = Statement.newBuilder(statementString)
                 .bind("workspaceId")
                 .to(workspaceId)
                 .build();
         try (ResultSet resultSet = dbClient.singleUse().executeQuery(statement)) {
             while (resultSet.next()) {
-                if (filterType.equals("status")){
-                    if (!resultSet.isNull("status")) {
-                        filterList.add(resultSet.getString("status"));
-                    }
-                }
-                else if (filterType.equals("location")){
-                    if (!resultSet.isNull("location")) {
-                        filterList.add(resultSet.getString("location"));
-                    }
-                }
-                else { // the filterType is kind
-                    if (!resultSet.isNull("kind")) {
-                        filterList.add(resultSet.getString("kind"));
-                    }
+                if (!resultSet.isNull(filterType)) {
+                    filterList.add(resultSet.getString(filterType));
                 }
             }
         }
         return filterList;
     }
 
     /**
-     * This is used to execute a query that was built dynamically and column types are unknown and return a list of data
-     * @param statement - A statement that holds the query
-     * @param dbClient - A client for connection to the DB
+     * This is used to execute a query that was built dynamically and column types are unknown
+     * and return a list of data
+     *
+     * @param statement   - A statement that holds the query
+     * @param dbClient    - A client for connection to the DB
      * @param columnNames - A list of column names to use for getting real types
      * @param columnTypes - A list of column types to use with Arrays
-     * @return a list os lists where each row holds information of one asset in strings (to use in template)
+     * @return a list os lists where each row holds information of one asset in strings (to use
+     * in template)
      */
-    public List<List<String>> executeQueryAndReturnList(Statement statement, DatabaseClient dbClient, List<String> columnNames,  List<String> columnTypes){
+    public List<List<String>> executeQueryAndReturnList(Statement statement,
```

Comment:
While this technically describes what the function does, the naming is a bit too general. it doesn't execute just any query - it executes the `getAllAssets` query and returns a list of them. The method name should reflect that - something like `getAllAssetsList` or `executeGetAllAssets` etc.

Reply:
It actually does execute all of the queries, all of them use this function.